presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetastoreClosure.java

```diff
@@ -171,6 +183,22 @@ public void dropColumn(HiveIdentity identity, String databaseName, String tableN
         return delegate.getPartitionNamesByParts(identity, databaseName, tableName, parts);
     }
 
+    private List<Partition> getExistingPartitionsByNames(HiveIdentity identity, Table table, List<String> partitionNames)
+    {
+        Map<String, Partition> partitions = delegate.getPartitionsByNames(identity, table, partitionNames).entrySet().stream()
+                .peek(entry -> {
+                    // make sure every partition is present
+                    if (!entry.getValue().isPresent()) {
+                        throw new PartitionNotFoundException(table.getSchemaTableName(), extractPartitionValues(entry.getKey()));
```

Comment:
If we saw partitions `[1, 2, ..., 1000]` earlier during planning and now `50` has been deleted in the metastore, should we fail the query?
related, do we want Presto to be able to query a table where partitions are constantly added and removed? (think: ACID without Hive ACID)

Reply:
> do we want Presto to be able to query a table where partitions are constantly added and removed? 

adding shouldn't be a problem. Removing makes query fail.

It used to fail before, but in `CachingHiveMetastore#loadPartitionColumnStatistics`. This only makes it explicit.

Reply:
We have to fail, because other code might be relying on it to exist. If partitions are constantly removed, the query will break anyway, since splits will be generated for files that won't exist when the split is used.